pkg/tests/mappings/mappings_test.bzl

```diff
@@ -974,3 +975,53 @@ def mappings_unit_tests():
         "mappings_unit_tests",
         strip_prefix_test,
     )
+
+def _gen_manifest_test_main_impl(ctx):
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = ctx.outputs.out,
+        substitutions = {
+            "${EXPECTED}": ctx.attr.expected,
+            "${TARGET}": ctx.attr.target,
+            "${TEST_NAME}": ctx.attr.test_name,
+        },
+    )
```

Comment:
I've been meaning to publish something WRT https://github.com/bazelbuild/bazel-skylib/issues/191, which would obviate this setup, but never got around to it.

Reply:
ack.